lib/resources/majors.tsx

```diff
@@ -572,14 +572,263 @@ const baCogSci: Major = {
+          required_credits: 1,
+          criteria: 'AS.173.115[C]^OR^AS.173.111[C]',
+        },
+        {
+          description:
+            '<b>General Physics II / Electricity and Magnetism I</b> <br /> Select one of the following: <br /> AS.171.106 Electricity and Magnetism I' +
+            '<br /> AS.171.102 General Physics: Physical Science Major II <br />AS.171.104 General Physics/Biology Majors II' +
+            '<br /> AS.171.108 General Physics for Physical Science Majors (AL)',
+          required_credits: 4,
+          criteria:
+            'AS.171.106[C]^OR^AS.171.102[C]^OR^AS.171.104[C]^OR^AS.171.108[C]',
+        },
+        {
+          description:
+            '<b>General Physics Laboratory II / Electricity and Magnetism Laboratory</b> <br /> AS.173.116 Electricity and Magnetism Laboratory' +
+            '<br /> <i>OR</i> <br /> AS.173.112 General Physics Laboratory II',
+          required_credits: 1,
+          criteria: 'AS.173.116[C]^OR^AS.173.112[C]',
+        },
+        {
+          description:
+            '<b>Calculus I</b> <br />' +
+            'AS.110.106 Calculus I (Biology and Social Sciences)' +
+            '<br /> <i>OR</i> <br />' +
+            'AS.110.108 Calculus I (Physical Sciences and Engineering)',
+          required_credits: 4,
+          criteria: 'AS.110.106[C]^OR^AS.110.108[C]',
+        },
+        {
+          description:
+            '<b>Calculus II</b> <br />' +
+            'AS.110.109 Calculus II (For Physical Sciences and Engineering)' +
+            '<br /> <i>OR</i> <br />' +
+            'AS.110.107 Calculus II (For Biological and Social Science)' +
+            '<br /> <i>OR</i> <br />' +
+            'AS.110.113 Honors Single Variable Calculus',
+          required_credits: 4,
+          criteria: 'AS.110.107[C]^OR^AS.110.109[C]^OR^AS.110.113[C]',
+        },
+      ],
+    },
+    {
+      name: 'Advanced Elective Courses',
+      required_credits: 3,
+      min_credits_per_course: 1,
+      description:
+        'Students must take three credits of advanced chemistry courses beyond AS.030.305-AS.030.306. ' +
+        'None of the advanced course requirements may be fulfilled with research. ' +
+        'Students must also take nine credits of advanced chemistry courses, or science electives at the 300-level or higher ' +
+        'approved by a Department of Chemistry advisor, and/or mathematics beyond Calculus II',
+      criteria:
+        'AS Chemistry[D]^AND^(200[L]^OR^300[L])^OR^AS Mathematics[D]^AND^(200[L])^NOT^AS.030.305[C]^NOT^AS.030.306[C]',
+      fine_requirements: [
+        {
+          description:
+            '<b>Advanced chemistry courses</b> <br /> Three credits of advanced chemistry courses beyond AS.030.305-AS.030.306, ' +
+            'cannot be fulfilled with research',
+          required_credits: 3,
+          criteria:
+            'AS Chemistry[D]^AND^(200[L]^OR^300[L])^NOT^AS.030.305[C]^NOT^AS.030.306[C]',
```

Comment:
I believe advanced chemistry courses is upper level (300 or 400 level). Also, I think instead of NOT, maybe use the exclude property of the javascript object. Refer to the documentation of how exclude works, but it is formatted the same way as a criteria string. Try adding (and testing) the line:

exclude: "AS.030.305[C]^OR^AS.030.306[C]",

Otherwise, I think the criteria string for this fine requirement looks good.